sansio_lsp_client/structs.py

```diff
@@ -61,6 +61,10 @@ class Position(BaseModel):
     line: int
     character: int
 
+    # for sorting
+    def __lt__(self, other):
+        return (self.character < other.character) if self.line == other.line else (self.line < other.line)
```

Comment:
Use the fact that `(line, column)` tuples sort nicely:

```suggestion
        return (self.line, self.character) < (other.line, other.character)
```

Also, doesn't this need `__gt__` and other methods too? `@functools.total_ordering`?

Reply:
I didn't want to create new objects. I'm using this to sort 'Format Document' edits, and it can be a large list. I'll check how it affects performance.

> Also, doesn't this need `__gt__` and other methods too? `@functools.total_ordering`?

[Here](https://docs.python.org/3/howto/sorting.html#odd-and-ends) the docs say:
> The sort routines are guaranteed to use __lt__() when making comparisons between two objects. So, it is easy to add a standard sort order to a class by defining an __lt__() method